modules/pam_namespace/pam_namespace.c

```diff
@@ -664,13 +727,16 @@ static int parse_config_file(struct instance_data *idata)
      * process_line to process each line.
      */
 
-    memset(&globbuf, '\0', sizeof(globbuf));
-    oldlocale = setlocale(LC_COLLATE, "C");
-    glob(NAMESPACE_D_GLOB, 0, NULL, &globbuf);
-    if (oldlocale != NULL)
-	setlocale(LC_COLLATE, oldlocale);
-
```

Comment:
The old code did change the locale to "C" to have a reproduceable output.
The new code does not have this. Since the glob() output depends on LC_COLLATE, we should set the locale again to make sure that no surprises happen.

Reply:
Sorry, I see that we have an additional "sort" with "strcmp".
So the result can be different than before, as glob() uses strcoll() or something similar internal.
If admins only use ASCII characters it could only become a problem with spaces or '-', not sure here. If admins uses UTF-8 characters, then the result may be surprising. But as we used "C" and not "C.UTF-8" already before, filenames with UTF-8 are not really well handled anyways.

@ldv-alt @t8m what do you think?

I see the following options:
- keep the current strcmp() implementation, which always returns something reproducible, but may be different to before. Advantage is, we don't need to mess around with setlocale() and locale handling
- use strcoll() and set locale to "C"
- use strcoll() and try to set locale to "C.UTF-8" and if that does not exist, fallback to "C".

Reply:
The manual page states that `In the POSIX or C locales strcoll(3) is equivalent to strcmp(3)`.
If that claim is correct, then we can safely use strcmp.

Reply:
I cannot find that in my versions of strcoll manual pages nor in the POSIX version. But with the C locale we sort in byte order, which matches strcmp, so we seem to be fine here.